### PR TITLE
redirect from old aleph_id to new alma id

### PR DESCRIPTION
**Pull request by antmoth**
Addresses ticket # SEARCH-1358.

This does technically work, I think, but I don't like it and it's messy. I'm also not able to fully confirm that the holdings ultimately work, which is fine, because my workaround is terrible. WIP.

Pairs with a redirect on search's end, not a PR yet: https://github.com/mlibrary/search/tree/SEARCH-1358-redirect-aleph-links

### CODE REVIEW

**Comment by billdueber** [comment]
If it works, it works, right?

First off, there's something almost identical happening with Summon in the `#initialize`, which makes me think that's the place for this process. We should at least just pick one or the other. 

Second, I'm a little worried that we don't know for sure that no other foci have a nine-digit ID -- should probably verify that the datastore is the catalog (much like the summon code in `#initialize`)

I don't understand the need for the `do_holdings` bit (not that I'm of the opinion that it's unnecessary, just that I don't know anything about it). Quick primer?

<here I cut all sorts of crap about doing a aleph-to-alma-id, given that Tim just said in slack that it's not really reliable>

So...I think just verify that it's the catalog and move the code?

**Comment by antmoth** [comment]
The do_holdings bit is because if I don't do _something_ there, a chain of events causes a nil-where-there-shouldn't-be-one error, on account of the holdings is still using the `aleph_id` as the query text for _id_ which of course brings up nothing. So I can probably do something better there.

File: lib/spectrum/bib_record.rb [hunk 1]
```diff
@@ -16,9 +16,13 @@ class BibRecord
       'MX' => false,
     )
 
-    def self.fetch(id:, url:, rsolr_client_factory: lambda{|url| RSolr.connect(url: url)}, escaped_id: RSolr.solr_escape(id))
+    def self.fetch(id:, url:, id_field: 'id', rsolr_client_factory: lambda{|url| RSolr.connect(url: url)}, escaped_id: RSolr.solr_escape(id))
       client = rsolr_client_factory.call(url)
-      BibRecord.new(client.get('select', params: { q: "id:#{escaped_id}"}))
+      # extracting this variable instead of running in the new()
+      # makes the difference on my machine between it working and not
+      # I don't know why and I don't like that I don't know why
```
Comment (antmoth) [inline lib/spectrum/bib_record.rb:26]: Particularly concerned by this

File: lib/spectrum/request/record.rb [hunk 2]
```diff
@@ -22,6 +22,13 @@ def initialize(request)
         @request = request
         if request.params[:source] == 'summon'
           @query = "#{@request.params['id_field']}:#{unfiltered_id(request)}"
+        elsif request.params[:source] == 'mirlyn'
+          id_field = @request.params['id_field']
```
Comment (billdueber) [inline lib/spectrum/request/record.rb:26]: Is params['id_field'] always set? If not, you run the risk of sending a `nil`, which will in fact override the keyword argument in fetch. You may want/need to explicitly `id_field ||= 'id'` in the body of `Holdings#fetch` if you're not positive `nil` will never come in.

Comment (antmoth) [reply lib/spectrum/request/record.rb:26]: The existing code seems to assume here that `params['id_field']` is always set (lines 24 and 33 here, or 24 and 26 in the `primo` branch). I also do have `request[:id_field] || 'id'` in the body of Request::Holdings#initialize, which is the `request` that is being passed in to `BibRecord#fetch`.

File: lib/spectrum/request/holdings.rb [hunk 3]
```diff
@@ -3,11 +3,16 @@
 module Spectrum
   module Request
     class Holdings
-      attr_reader :id, :focus
+      attr_reader :id, :focus, :id_field
 
       def initialize(request)
```
Comment (billdueber) [inline lib/spectrum/request/holdings.rb:8]: I might simplify this to something like (untested!):

```ruby
 
def initialize(request)
  @request = request
  source = @request.params['source']
  id = case source
       when 'summon', 'mirlyn'
         unfiltered_id(@request)
       else
         RSolr.solr_escape(unfiltered_id(request))
       end
  id_field = case source
             when 'mirlyn'
               if id.length == 9
                 'aleph_id'
               else
                 @request.params['id_field']
               end
             end
  @query = "#{id_field}:#{id}"
end 

```

Comment (antmoth) [reply lib/spectrum/request/holdings.rb:8]: Do we not want the solr_escaped id for mirlyn? Previously mirlyn would have been getting the solr_escaped id.

**Pull request merged**